### PR TITLE
EZP-32240: Commerce API revision

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EZP-32240
| Versions      | 3.x, master

REST endpoints are covered in https://github.com/ezsystems/ezplatform-rest-reference/pull/12

~Note: Merge after https://github.com/ezsystems/developer-documentation/pull/1335 to combine with new FT reference structure~

### CODE REVIEW

**Review by Steveb-p — COMMENTED** [review]
Looks good at first glance, I'll have a second pass on reading this later :+1:

File: docs/api/shop_business_api.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        'language' => 'de_DE',
+        'filterType' => 'productList',
+    )
+);
+
+/** @var OutputLoadList $output */
+$output = $this->getBusinessApi()->call('catalog.load_products', $input);
+
+$html = $this->renderView(
+    'SilversolutionsEshopBundle:Catalog:listProductNodes.html.twig',
```
Comment (Steveb-p) [inline docs/api/shop_business_api.md:111]: This render call uses deprecated (in Symfony 5, removed) template reference. 

You probably wanted `@SilversolutionsEshopBundle/Catalog/listProductNodes.html.twig` or `@ezdesign/Catalog/listProductNode.html.twig`.

**Review by Steveb-p — APPROVED** [review]

File: docs/api/field_types_reference/sesselection.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# SesSelection
+
+This Field Type stores a single selection choice. 
+
+Unlike the [Selection Field Type](selectionfield.md),
```
Comment (dabrt) [inline docs/api/field_types_reference/sesselection.md:5]: Which piece of information is more important here? I'd go with "The SesSelection Field Type is configured in a YAML file, unlike the [Selection Field Type](selectionfield.md)."
This way we shift emphasis.

File: docs/api/field_types_reference/sesselection.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+the SesSelection Field Type is configured in a YAML file.
+For that reason, it is possible to set up SiteAccess-specific selection Field Types.
```
Comment (dabrt) [inline docs/api/field_types_reference/sesselection.md:7]: "for that reason, it is possible", or "this way you can..."?

File: docs/api/field_types_reference/sesselection.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+siso_core.default.sesselection.news_type:
+    default: general
+    translation_context: news
+    options:
+        general: general
+        sports: sports
+        culture: culture
+```
+
+The `translation_context` key is optional.
```
Comment (dabrt) [inline docs/api/field_types_reference/sesselection.md:21]: What is translation context?

File: docs/api/field_types_reference/specificationstype.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# SpecificationsType
+
+The Field Type stores a structured list of attributes for products.
```
Comment (dabrt) [inline docs/api/field_types_reference/specificationstype.md:3]:
```suggestion
This Field Type stores a structured list of attributes for products.
```

File: docs/api/shop_business_api.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
+# Shop business API
+
+Shop business API is the layer between the application entry points (like controllers or CLI commands) and the particular shop-related services.
+
+To access the Shop business API, you have to use the Business API invocation service.
+This service is the access point to the Business API and defined by a service with the ID `ses_eshop.business_api.invocation`.
```
Comment (dabrt) [inline docs/api/shop_business_api.md:6]: "is defined"?

File: docs/api/shop_business_api.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+## addProducts
+
+`basket.add_products` adds one or more products to the basket.
+
+### Example
+
+``` php
+$outputGetBasket = $this->getBusinessApi()->call('basket.get_basket', $inputGetBasket);
+//clear all messages for each request
```
Comment (dabrt) [inline docs/api/shop_business_api.md:44]:
```suggestion
// Clear all messages for each request
```

File: docs/api/shop_business_api.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+$outputGetBasket->basket->clearAllMessages();
+$itemData = new ItemData(
+    array(
+        'quantity'    => '1',
+        'isVariant'   => '',
+        'variantCode' =>  '',
+        'sku'         => '1000',
+    )
+);
+/** @var InputAddItemToBasket $inputAddItemToBasket */
```
Comment (dabrt) [inline docs/api/shop_business_api.md:54]:
```suggestion

/** @var InputAddItemToBasket $inputAddItemToBasket */
```

File: docs/api/shop_business_api.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+$inputAddItemToBasket = new InputAddItemToBasket(
+   array(
+       'itemData' => $itemData,
+       'basket'   => $outputGetBasket->basket,
+   )
+);
+try {
```
Comment (dabrt) [inline docs/api/shop_business_api.md:61]:
```suggestion

try {
```

File: docs/api/shop_business_api.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    $outputGetBasket = $this->getBusinessApi()->call('basket.add_products', $inputAddItemToBasket);
+} catch (\InvalidArgumentException $e) {
+    // ....
+}
+$message = $this->getBasketMessage($outputGetBasket->basket);
+
```
Comment (dabrt) [inline docs/api/shop_business_api.md:67]:
```suggestion
```

File: docs/api/shop_business_api.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+       'itemData' => $itemData,
+       'basket'   => $outputGetBasket->basket,
+   )
+);
+try {
+    $outputGetBasket = $this->getBusinessApi()->call('basket.add_products', $inputAddItemToBasket);
+} catch (\InvalidArgumentException $e) {
+    // ....
+}
+$message = $this->getBasketMessage($outputGetBasket->basket);
```
Comment (dabrt) [inline docs/api/shop_business_api.md:66]:
```suggestion

$message = $this->getBasketMessage($outputGetBasket->basket);
```

File: docs/guide/catalog/catalog_api/routing_system.md [hunk 12]
```diff
@@ -66,7 +66,7 @@ The router is defined with the priority of 280, so you can still add your own ch
 
 ## Usage of the navigation service
 
-In order to determine whether a URL belongs to the catalog or [silver module](../../../api/commerce_api/field_type_reference/silver.module.md),
+In order to determine whether a URL belongs to the catalog or silver module,
```
Comment (dabrt) [inline docs/guide/catalog/catalog_api/routing_system.md:69]:
```suggestion
To determine whether a URL belongs to the catalog or a silver module,
```

File: docs/guide/catalog/product.md [hunk 13]
```diff
@@ -49,3 +49,27 @@ You also need to configure the Content Type to be treated as `createOrderablePro
 ``` yaml
 parameters.silver_eshop.default.catalog_factory.<content_type_identifier>: createOrderableProductNode
 ```
+
+## Prouct specifications
```
Comment (dabrt) [inline docs/guide/catalog/product.md:53]:
```suggestion
## Product specifications
```

File: docs/guide/catalog/product.md [hunk 14]
```diff
@@ -49,3 +49,27 @@ You also need to configure the Content Type to be treated as `createOrderablePro
...
+
+You can configure the available product specifications and default values using the following configuration:
```
Comment (dabrt) [inline docs/guide/catalog/product.md:55]:
```suggestion
You can configure the available product specifications and default values by using the following configuration:
```

File: docs/guide/catalog/product.md [hunk 15]
```diff
@@ -49,3 +49,27 @@ You also need to configure the Content Type to be treated as `createOrderablePro
...
+                value: ""
+                options: ['mm','cm', 'in']
+            -
+                id: diameter
+                label: "Diameter"
+                value: ""
+                options: ['mm','cm']
+```
+
+The optional `option` attribute enables you to add a select field offering e.g. a selection of units.
```
Comment (dabrt) [inline docs/guide/catalog/product.md:75]:
```suggestion
With the optional `option` attribute you can add a select field that offers, for example, a selection of units.
```

**Review by dabrt — APPROVED** [review]
I am thinking whether we should not apply `code` styling to all mentions of fieldnames (`SpecificationsType`, `TextBlock`) when appearing in paragraphs or tables.

Now we tend to use code styling in tables and outside of tables we use plain styling. 
I'd go with plain styling in page headers only.

Food for thought.

File: docs/api/field_type_reference.md [hunk 16]
```diff
@@ -39,11 +39,16 @@ Custom Field Types have to be programmed in PHP. However, the built-in Field Ty
 | [RelationList](field_types_reference/relationlistfield.md) | Validates and stores a list of relations to Content items. | Yes, with [`FieldRelation` Criterion](../guide/search/criteria_reference/fieldrelation_criterion.md) | Yes |
 | [RichText](field_types_reference/richtextfield.md) | Validates and stores structured rich text in DocBook xml format, and exposes it in several formats. Available via [RichTextBundle](https://github.com/ezsystems/ezplatform-richtext). | Yes[^1^](#1-note-on-legacy-search-engine)  | Yes |
 | [Selection](field_types_reference/selectionfield.md) | Validates and stores a single selection or multiple choices from a list of options. | Yes[^1^](#1-note-on-legacy-search-engine) | Yes |
+| [SesExternaldata](field_types_reference/sesexternaldata.md) | Uses external storage to store data. |||
+| [SesProfiledata](field_types_reference/sesprofiledata.md) | Stores address data for a customer. | No | No |
+| [SesSelection](field_types_reference/sesselection.md) | Stores a single selection choice using options from a YAML file. | Yes | Yes |
```
Comment (DominikaK) [inline docs/api/field_type_reference.md:44]:
```suggestion
| [SesSelection](field_types_reference/sesselection.md) | Stores a single selection choice based on options from a YAML file. | Yes | Yes |
```

File: docs/api/field_types_reference/sesselection.md [hunk 17]
```diff
@@ -0,0 +1,23 @@
+# SesSelection
+
+This Field Type stores a single selection choice. 
+
+Unlike the [Selection Field Type](selectionfield.md),
+the SesSelection Field Type is configured in a YAML file.
```
Comment (DominikaK) [inline docs/api/field_types_reference/sesselection.md:6]:
```suggestion
The SesSelection Field Type is configured in a YAML file,
unlike the [Selection Field Type](selectionfield.md).
```

File: docs/api/field_types_reference/sesselection.md (hunk 3 above)
Comment (DominikaK) [inline docs/api/field_types_reference/sesselection.md:7]:
```suggestion
This way, you can set up SiteAccess-specific selection Field Types.
```

File: docs/api/field_types_reference/sesselection.md (hunk 4 above)
Comment (DominikaK) [inline docs/api/field_types_reference/sesselection.md:21]:
```suggestion
The `translation_context` key, which identifies the context used for translating the labels, is optional. 
```

File: docs/api/shop_business_api.md (hunk 6 above)
Comment (DominikaK) [inline docs/api/shop_business_api.md:6]:
```suggestion
This service is the access point to the Business API and is defined by a service with the ID `ses_eshop.business_api.invocation`.
```

**Pull request merged**